### ISSUE TITLE
Search only works in Firefox incognito

### ISSUE THREAD

**Issue by atalii** [NixOS/nixos-search#699]
If I load up `search.nixos.org` on my browser, any attempt to search results in `Uncaught TypeError: n is undefined` in bundle.js. Interestingly, this issue goes away if I open the same in incognito mode, but persists if I disable the cache via the network tab in FF. Any ideas as to what's going on? Thanks!

**Comment by hraban** [comment]
fwiw cannot reproduce

**Comment by atalii** [comment]
Yeah, I didn't expect it to be easily reproducible. When I load the page outside of incognito, the console prints a warning that the page is compiled in dev mode. In the network inspector, there's no difference in which assets are queried nor does there seem to be any difference in the response. The error outside of incognito seems to have changed though: `TypeError: e.llRecoveryInstitute is undefined`.

Disabling and enabling all my extensions one-by-one to pinpoint a cause doesn't show anything either. Honestly, I'm quite mystified. I have no idea what else varies between incognito and normal.

**Comment by atalii** [comment]
The problem fixed itself some time over the past few months. Feel free to close the issue (especially if no one else has experienced this). Thanks!

**Comment by ugur-a** [comment]
Hi, I seem to have the same problem. There is one small difference though: I only have the first error (`Uncaught TypeError: n is undefined`). It appears one time on each keystroke, and when I hit search it starts repeating endlessly.

The search results are actually there, by the way. I can see them in Developer Mode -> Network -> Response -> hits

I've been having this problem for something like a month now but haven't seen anyone taking about this. 

If anyone has the same problem and stumbles across this issue, I've been using [nix-search-cli](https://github.com/peterldowns/nix-search-cli) and can recommend it.

**Comment by gihrig** [comment]
Experienced this today.
macOS Monterey 12.7.5
Firefox 126.0

I opened `https://search.nixos.org/packages` and searched for "beyondcompare" results page loaded, a few seconds later went blank.
Page info (lock icon > Connection secure > Security tab "is this website storing information on my computer": Yes 9 bytes

"Clear Cookies and site data" fixed the issue for me. Also after repeated successful searches (including "beyondcompare") "is this website storing information on my computer": No

---

Later the problem reappeared. Still "is this website storing information on my computer": No

**Comment by gihrig** [comment]
A possible reproduction:
macOS Monterey 12.7.5
Firefox 126.0
System Preferences > General > Appearance "Dark" selected

1. Open a new tab and paste `https://search.nixos.org/packages` Enter. The normal search page appears.
2. Add a trailing slash `https://search.nixos.org/packages/` Enter. A blank white page appears.
3. Remove the trailing slash `https://search.nixos.org/packages` Enter. A blank black page with footer "Please help us improve the search ..." appears.
4. A few seconds later the footer disappears.

**Comment by atalii** [comment]
Can't reproduce with that on Linux w/ FF 125. Dark mode may be a common denominator, though.

**Comment by gihrig** [comment]
Same behavior with light auto or dark mode.
Safari displays blank screen from "https://search.nixos.org/packages/" but recovers as expected with "https://search.nixos.org/packages".

It seems odd that a trailing slash would cause a blank display. That is true in Safari as well, but in Safari the page display recovers when the trailing slash is removed.

**Comment by atalii** [comment]
The trailing slash issue isn't too hard, i don't think: The returned html asks for the relative path "bundle.js". If the user agent is on "https://search.nixos.org/packages", that becomes "https://search.nixos.org/bundle.js". Contrarily, if the trailing slash is given, the request for "https://search.nixos.org/packages/bundle.js" 404's. I'd expect the trailing slash version of the URL ("/packages/") to 404 in the first place, but maybe that's just a quirk of the hosting service... Strange how this may contribute to the issue.

**Comment by 12-G** [comment]
I have experienced the same problem. As you said, when you open the website in incognito mode, everything works. Weird
